### DISCUSSION TITLE
Self created language extension is not recognized

### DISCUSSION THREAD

**Discussion by FireballTHM** [microsoft/vscode-discussions#796] [Extension Development QnA] [answered]
I have created a language extension for an older programming language "FGL". So far I copied it into the folder USER\USERNAME\.vscode\extensions\fgl.
Earlier installations of VS Code recognized the extension without any problems. However, in more recent installations of VS CODE, it is no longer recognized and does not show up in the extensions.

Unfortunately, we are at a bit of a loss as to what this could be.

Many greetings

**Comment by r3m0t** [comment, marked as answer]
Use the `--install-extension` command line argument and pass the path to the vsix. Never fails

**Reply by FireballTHM** [reply]
Many thanks, The installation via VSIX works without problems.
And with the help of the article that gjsjhonmurray pointed out to me, I was able to pack the extension very quickly.

**Comment by FireballTHM** [comment]
Thank you very much for your reply.

I am currently trying to pack the language into my first vsix package. 
Unfortunately I have not yet found a complete guide on how to do this relatively easily.
If anyone here knows of a good post, I would be grateful.

Currently I've been copying the language into the extensions folder of the application, this works temporarily.

**Reply by gjsjohnmurray** [reply]
Did you find https://code.visualstudio.com/api/working-with-extensions/publishing-extension?

Once you have installed vsce it should be as simple as running this in your repo root folder:

```
vsce package
```

**Reply by FireballTHM** [reply]
No, even after hours of searching I did not find what I was looking for.

Many thanks, the article helped a lot.
